### PR TITLE
dev

### PR DESCRIPTION
**Pull request by soudblox**
## Type
- [x] Bug Fix: Fixed $randomEmoji
- [x] Functions: $getAuditLogs
- [ ] Callbacks: 
- [ ] Handlers: \
- [ ] Others: Add $getAuditLogs[limit;userID;action;guildID;format], add {tag} to globalUserLeaderboard

Dependencies (Third Party Modules) needed: 

Want a credit? Discord tag or other social media link: blox#8880

Referenced Issue:  #34

## Description:

Available placeholders for getAuditLogs custom format:
{executor.username} username of user that triggers the audit log
{executor.id} tag of user that triggers the audit log
{executor.mention} mention of user that triggers the audit log
{target.id} the id of the audit log target(can be channel/message/user id)
{action} the action

list of audit log actions can be found in:
https://discord.js.org/#/docs/main/stable/typedef/AuditLogAction

### CODE REVIEW

File: package/functions/funcs/getAuditLogs.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            action = `ALL`,
+            guildID = d.message.guild.id,
+            format = `{executor.username} - {target.id} - {action}`
+        ] = inside.splits
+
+        const guild = d.client.guilds.cache.get(guildID)
+
+        if (!guild) return d.error(`:x: Invalid guild ID in \`$getAuditLogs${inside}\``)
+
+        const user = userID ? d.client.users.cache.get(userID) : null
```
Comment (XzFirzal) [inline package/functions/funcs/getAuditLogs.js:16]: U might want to change it to fetch

Comment (soudblox) [reply package/functions/funcs/getAuditLogs.js:16]: alr i'll change it

File: package/functions/parser.js [hunk 2]
```diff
@@ -310,7 +306,7 @@ const Parser = {
   $rolePerms:
     "Returns the role key permissions.;$rolePerms[roleID;separator (optional)]",
   $songInfo:
-    "Displays information about the song that is being played.;$songInfo[property;position]",
+    "Displays information about the song that is being played.;$songInfo[property]",
```
Comment (XzFirzal) [inline package/functions/parser.js:309]: Might want to put newer parser.js, then add the function

Comment (soudblox) [reply package/functions/parser.js:309]: mk, i'll change that too

**Comment by soudblox** [comment]
i also fixed randomEmoji

File: package/functions/funcs/randomEmoji.js [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
- 
- if (emoji.deleted) return {code:d.command.code.replaceLast(`$randomEmoji${inside.total ? inside.total : ""}`, "");
- 
- return {code:d.command.code.replaceLast(`$randomEmoji${inside.total ? inside.total : ""}`, `<:${emoji.name}:${emoji.id}>`)
-}
+    const code = d.command.code
+    
+    const inside = d.unpack();
+    
+    var guildId = inside.splits;
```
Comment (XzFirzal) [inline package/functions/funcs/randomEmoji.js:6]:
```js
var guildId = inside.splits
```
?
did u mean inside.inside

**Comment by soudblox** [comment]
oop yea, give me a second

**Pull request merged**